src/main/java/nl/thehyve/ocdu/controllers/UploadSessionController.java

```diff
@@ -70,16 +68,36 @@ public String username() {
             UploadSession currentUploadSession = uploadSessionService.getCurrentUploadSession(session);
             UploadSession.Step _step;
             switch (step) {
-                case "mapping": _step = UploadSession.Step.MAPPING; break;
-                case "feedback-data": _step = UploadSession.Step.FEEDBACK_DATA; break;
-                case "subjects": _step = UploadSession.Step.SUBJECTS; break;
-                case "feedback-subjects": _step = UploadSession.Step.FEEDBACK_SUBJECTS; break;
-                case "events": _step = UploadSession.Step.EVENTS; break;
-                case "feedback-events": _step = UploadSession.Step.FEEDBACK_EVENTS; break;
-                case "pre-odm-upload": _step = UploadSession.Step.PRE_ODM_UPLOAD; break;
-                case "odm-upload": _step = UploadSession.Step.ODM_UPLOAD; break;
-                case "final": _step = UploadSession.Step.FINAL; break;
-                default: _step = UploadSession.Step.MAPPING; break;
+                case "mapping":
```

Comment:
In principle all of that could be replaced by Step.valueOf() with addition of handling invalid strings